test/support/data_case.ex

```diff
@@ -10,6 +11,15 @@ defmodule Guardian.DB.Test.DataCase do
     end
   end
 
+  setup_all do
+    tmp_path = tmp_path("priv/test/migrations")
+    File.mkdir_p!(tmp_path)
+    Mix.Task.run("guardian.db.gen.migration")
```

Comment:
Why would I want te calling this task every single time the setup happens and the migrate?

The only time you would need to migrate is that you are changing the schema of the `Token` which wouldn't happen that often at all. 

So basically you will need to do `MIX_ENV=test mix ecto.reset` for be able to reset your database but I wouldn't add time to the testing.

Could you please put some benchmark on this?

Reply:
As stated, initial PR. I think it's going to be better to use a stub Repo as Ecto does in their migration tests. I will refactor.

Reply:
@yordis I believe the intention is to run against the generator rather than a hard coded test migration file. Please correct me if I'm wrong @starbelly.

The setup_all runs only once for the case which we should be able to limit to a single test file so it should only run once.

Reply:
That is the itention, but as stated, I think it's going to be better to to set the test up how ecto does in their gen migration tests: 

```elixir
defmodule Repo do
    def __adapter__ do
      true
    end

    def config do
      [priv: "tmp/#{inspect(Ecto.Gen.Migration)}", otp_app: :ecto]
    end
  end
```